framework/include/utils/MooseTypes.h

```diff
@@ -146,12 +146,15 @@ typedef MooseArray<std::vector<VectorValue<Real>>> VectorVariableTestCurl;
 
 namespace Moose
 {
+
+#include "libmesh/ignore_warnings.h" // For Minimum GCC Compiler
 const SubdomainID ANY_BLOCK_ID = libMesh::Elem::invalid_subdomain_id - 1;
 const SubdomainID INVALID_BLOCK_ID = libMesh::Elem::invalid_subdomain_id;
 const BoundaryID ANY_BOUNDARY_ID = static_cast<BoundaryID>(-1);
 const BoundaryID INVALID_BOUNDARY_ID = libMesh::BoundaryInfo::invalid_id;
 const std::set<SubdomainID> EMPTY_BLOCK_IDS = {};
 const std::set<BoundaryID> EMPTY_BOUNDARY_IDS = {};
+#include "libmesh/restore_warnings.h"
```

Comment:
These 2 `#include`s are in weird place - inside a namespace and interleaved by global constants. What is the reason for it?

Reply:
Apparently old GCC (4.8.4) is complaining that these values aren't used with -Wall -Wextra when included from some MOOSE files. I'm not sure of a better way to make it shut up. If you'd like I could just put in specific compiler directives here, but I figured it would be easier to just use the growing libMesh compiler directive. 

Actually, even this may not work. I think @brianmoose is going to build on a machine with this configuration and take a look to see if this is the best solution or not.

Reply:
If this is what we want to go with, then I'd suggest to add a comment that explains it. It was not really obvious to me what was going on.